test/commands/extended/check_consistency_test.py

```diff
@@ -0,0 +1,226 @@
+    self.assertFilterErrors(
+      {
+        'tails': [TransactionHash(self.hash1)],
+        'foo': 'bar',
+      },
+
+      {
+        'foo': [f.FilterMapper.CODE_EXTRA_KEY],
+      },
+    )
+
+  def test_fail_tails_null(self):
+    """
+    ``tails`` is null.
+    """
+    self.assertFilterErrors(
+      {
+        'tails': None,
+      },
+
+      {
+        'tails': [f.Required.CODE_EMPTY],
+      },
+    )
+
+  def test_fail_tails_wrong_type(self):
+    """
+    ``tails`` is not an array.
+    """
+    self.assertFilterErrors(
+      {
+        # It's gotta be an array, even if there's only one hash.
+        'tails': TransactionHash(self.hash1),
+      },
+
+      {
+        'tails': [f.Type.CODE_WRONG_TYPE],
+      },
+    )
+
+  def test_fail_tails_empty(self):
+    """
+    ``tails`` is an array, but it is empty.
+    """
+    self.assertFilterErrors(
+      {
+        'tails': [],
+      },
+
+      {
+        'tails': [f.Required.CODE_EMPTY],
+      },
+    )
+
+  def test_fail_tails_contents_invalid(self):
+    """
+    ``tails`` is a non-empty array, but it contains invalid values.
+    """
+    self.assertFilterErrors(
+      {
+        'tails': [
+          b'',
+          True,
+          None,
+          b'not valid trytes',
+
+          # This is actually valid; I just added it to make sure the
+          #   filter isn't cheating!
+          TryteString(self.hash1),
+
+          2130706433,
+          b'9' * 82,
+        ],
+      },
+
+      {
+        'tails.0':  [f.Required.CODE_EMPTY],
+        'tails.1':  [f.Type.CODE_WRONG_TYPE],
+        'tails.2':  [f.Required.CODE_EMPTY],
+        'tails.3':  [Trytes.CODE_NOT_TRYTES],
+        'tails.5':  [f.Type.CODE_WRONG_TYPE],
+        'tails.6':  [Trytes.CODE_WRONG_FORMAT],
+      },
+    )
+
+
+class CheckConsistencyCommandTestCase(TestCase):
+  # noinspection SpellCheckingInspection
+  def setUp(self):
+    super(CheckConsistencyCommandTestCase, self).setUp()
+
+    self.adapter = MockAdapter()
+    self.command = CheckConsistencyCommand(self.adapter)
+
+    # Define some tryte sequences that we can re-use across tests.
+    self.milestone =\
+      TransactionHash(
+        b'TESTVALUE9DONTUSEINPRODUCTION99999W9KDIH'
+        b'BALAYAFCADIDU9HCXDKIXEYDNFRAKHN9IEIDZFWGJ'
+      )
+
+    self.hash1 =\
+      TransactionHash(
+        b'TESTVALUE9DONTUSEINPRODUCTION99999TBPDM9'
+        b'ADFAWCKCSFUALFGETFIFG9UHIEFE9AYESEHDUBDDF'
+      )
+
+    self.hash2 =\
+      TransactionHash(
+        b'TESTVALUE9DONTUSEINPRODUCTION99999CIGCCF'
+        b'KIUFZF9EP9YEYGQAIEXDTEAAUGAEWBBASHYCWBHDX'
+      )
+
+  def test_wireup(self):
+    """
+    Verify that the command is wired up correctly.
+    """
+    self.assertIsInstance(
+      Iota(self.adapter).checkConsistency,
+      CheckConsistencyCommand,
+    )
+
+  def test_happy_path(self):
+    """
+    Successfully checking consistency.
+    """
+
+    self.adapter.seed_response('checkConsistency', {
+      'state': True,
+    })
```

Comment:
🔖  (adding a bookmark here in case `check_consistency` is actually supposed to return a list of booleans; see comment above)